apps/opik-python-backend/tests/test_metrics_factory.py

```diff
@@ -160,3 +161,91 @@ def test_levenshtein_metric_no_match(self):
         assert result.value == 0.0
 
 
+class TestMetricReasons:
+    """Tests for metric reason fields (required for hierarchical_reflective optimizer)."""
+
+    def test_equals_metric_includes_reason_on_match(self):
+        """Test equals metric includes reason field on match."""
+        metric_fn = MetricFactory.build("equals", {}, "model")
+        dataset_item = {"answer": "test"}
+        result = metric_fn(dataset_item, "test")
+        
+        assert result.reason is not None
+        assert "match" in result.reason.lower()
+
+    def test_equals_metric_includes_reason_on_no_match(self):
+        """Test equals metric includes reason field on no match."""
+        metric_fn = MetricFactory.build("equals", {}, "model")
+        dataset_item = {"answer": "test"}
+        result = metric_fn(dataset_item, "different")
+        
+        assert result.reason is not None
+        assert "no match" in result.reason.lower()
+
+    def test_levenshtein_metric_includes_reason(self):
+        """Test levenshtein metric includes reason field with similarity percentage."""
+        metric_fn = MetricFactory.build("levenshtein_ratio", {}, "model")
+        dataset_item = {"answer": "hello"}
+        result = metric_fn(dataset_item, "hallo")
+        
+        assert result.reason is not None
+        assert "similarity" in result.reason.lower()
+        assert "%" in result.reason
+
+
+class TestTemplateSyntaxConversion:
+    """Tests for template syntax conversion from {{var}} to {var}."""
+
+    def test_convert_single_variable(self):
+        """Test converting single variable."""
+        result = _convert_template_syntax("Hello {{name}}")
```

Comment:
Minor: it's not a good practice to test a private function. If it's a testable unit, it's better to move it to a dedicated module and make it public there.